bsoid_app/video_creator.py

```diff
@@ -20,87 +19,83 @@ def selected_vid(vid_file):
     return vid_file
 
 
-class creator:
+class Creator:
 
-    def __init__(self, root_path, data_directories, processed_input_data,
-                 pose_chosen, working_dir, prefix, framerate, clf, input_filenames):
-        st.subheader('GENERATE VIDEOS SNIPPETS FOR INTERPRETATION')
-        self.root_path = root_path
-        self.data_directories = data_directories
+    def __init__(self, ROOT_PATH, DATA_DIRECTORIES, processed_input_data, pose_chosen, file_directory, PREFIX, FRAMERATE, MIN_TIME, FTYPE, NUMBER_EXAMPLES, PLAYBACK_SPEED, clf, input_filenames):
+        print('GENERATE VIDEOS SNIPPETS FOR INTERPRETATION')
+        self.root_path = ROOT_PATH
+        self.data_directories = DATA_DIRECTORIES
         self.processed_input_data = processed_input_data
         self.pose_chosen = pose_chosen
-        self.working_dir = working_dir
-        self.prefix = prefix
-        self.framerate = framerate
+        self.prefix = PREFIX
+        self.min_time = MIN_TIME
+        self.framerate = FRAMERATE
+        self.playback_speed = PLAYBACK_SPEED
         self.clf = clf
         self.input_filenames = input_filenames
-        self.file_directory = []
+        self.file_directory = file_directory
         self.d_file = []
         self.vid_dir = []
         self.vid_file = []
         self.frame_dir = []
-        self.filetype = []
+        self.filetype = FTYPE
         self.width = []
         self.height = []
         self.bit_rate = []
         self.num_frames = []
         self.avg_frame_rate = []
         self.shortvid_dir = []
         self.min_frames = []
-        self.number_examples = []
+        self.number_examples = NUMBER_EXAMPLES
```

Comment:
So `number_example` is supposed to be int according to your documentation. But here they were using an array. Does this actually work?

Reply:
yes as far as I can tell this is not creating any issues 🤷‍♀️

Reply:
self.number_examples = st.slider('Select number of non-repeated examples', 1, 20, 5)

this is from the original app. You select a number from a slider, but in the end it is just an integer number of examples?